reactioncommerce/reaction
# Add moveOrderItems mutation

Part of #4999
Impact: **minor**  
Type: **feature**

## Changes
A new `moveOrderItems` mutation allows you to move one or more order items from one fulfillment group to another existing fulfillment group on the same order.

The operator UI does not yet implement this.

- `moveOrderItems` can be called internally to synchronize from an external system by setting `context.isInternalCall` to `true`
- `moveOrderItems` can be called through GraphQL by any user with "orders" permission for the shop that owns the order
- `moveOrderItems` can be called through GraphQL by the user who placed the order. In this case, the move will only succeed if both the item status and the order status are "new".

## Breaking changes
None

## Testing
1. Place an order.
1. Verify that the mutation works as described.
1. Verify that you can move order items when authenticated as a user with "orders" role or as the user who placed the order, but not when authenticated as someone else or when not authenticated.
